lib/puppet/pops/types/p_timespan_type.rb

```diff
@@ -65,9 +65,9 @@ def convert_arg(arg, min)
       when String
         impl_class.parse(arg)
       when Integer
-        arg == impl_class.new(arg * Time::NSECS_PER_SEC)
+        impl_class.new(arg * Time::NSECS_PER_SEC)
       when Float
-        arg == (min ? -Float::INFINITY : Float::INFINITY) ? arg : impl_class.new(arg * Time::NSECS_PER_SEC)
+        (min ? -Float::INFINITY : Float::INFINITY) ? arg : impl_class.new(arg * Time::NSECS_PER_SEC)
```

Comment:
I'm a little confused here, aren't both `-Float::INFINITY` and `FLoat::INFINITY` truthy? Will we ever reach the false branch of the outer ternary?

Reply:
Hm... good find - I did not really look close enough at the actual expression (doh) as I just removed the assignment - that looks completely wrong... fixing

Reply:
Fixed, and added negative tests.